File: ObjCVerify/ObjCVerify/Examples/Controller/TableViewCell侧滑/WYTestSideslipCellController.m
//
//  WYTestSideslipCellController.m
//  ObjCVerify
//
//  Created by guanren on 2026/1/15.
//

#import "WYTestSideslipCellController.h"
#import <Masonry/Masonry.h>
#import <WYBasisKitObjC/WYBasisKitObjC.h>

@interface WYTestSideslipCellController () <UITableViewDelegate, UITableViewDataSource>

@property (nonatomic, strong) UITableView *tableView;
@property (nonatomic, strong) NSMutableArray<NSString *> *dataSource;

// 控制开关
@property (nonatomic, assign) BOOL enableLongPull;
@property (nonatomic, assign) WYSideslipGesturePriority currentGesturePriority;

@end

@implementation WYTestSideslipCellController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.title = @"侧滑功能验证";
    self.view.backgroundColor = UIColor.magentaColor;
    
    [self setupNavigationBar];
    [self setupTableView];
    [self setupData];
    
    // 启用自动关闭侧滑功能（只需要调用一次）
    [UITableView wy_enableAutoCloseSideslip];
}

- (void)setupNavigationBar {
    // 添加长拉功能开关
    UIBarButtonItem *longPullButton = [[UIBarButtonItem alloc] initWithTitle:self.enableLongPull ? @"长拉:开" : @"长拉:关"
                                                                       style:UIBarButtonItemStylePlain
                                                                      target:self
                                                                      action:@selector(toggleLongPull)];
    
    // 添加手势优先级切换
    UIBarButtonItem *gesturePriorityButton = [[UIBarButtonItem alloc] initWithTitle:[self gesturePriorityTitle]
                                                                              style:UIBarButtonItemStylePlain
                                                                             target:self
                                                                             action:@selector(switchGesturePriority)];
    
    self.navigationItem.rightBarButtonItems = @[longPullButton, gesturePriorityButton];
}

- (void)setupTableView {
    self.tableView = [[UITableView alloc] initWithFrame:CGRectMake(10, UIDevice.wy_navViewHeight, UIDevice.wy_screenWidth - 20, UIDevice.wy_screenHeight - UIDevice.wy_navViewHeight) style:UITableViewStylePlain];
    self.tableView.delegate = self;
    self.tableView.dataSource = self;
    self.tableView.backgroundColor = UIColor.whiteColor;
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:@"Cell"];
    self.tableView.tableFooterView = [[UIView alloc] init];
    
    [self.view addSubview:self.tableView];
}

- (void)setupData {
    [self.dataSource removeAllObjects];
    for (NSInteger i = 1; i <= 20; i++) {
        [self.dataSource addObject:[NSString stringWithFormat:@"测试单元格 %ld", (long)i]];
    }
    [self.tableView reloadData];
}

- (NSMutableArray<NSString *> *)dataSource {
    if (!_dataSource) {
        _dataSource = [NSMutableArray array];
    }
    return _dataSource;
}

- (WYSideslipGesturePriority)currentGesturePriority {
    return _currentGesturePriority ?: WYSideslipGesturePriorityAutoSelection;
}

- (void)toggleLongPull {
    self.enableLongPull = !self.enableLongPull;
    
    // 重新创建导航栏按钮数组
    [self setupNavigationBar];
    
    // 使用封装的方法重置所有cell状态
    [self.tableView wy_resetAllVisibleCellsSideslipState];
    [self.tableView reloadData];
    
    wy_print(@"长拉功能: %@", self.enableLongPull ? @"开启" : @"关闭");
}

- (void)switchGesturePriority {
    switch (self.currentGesturePriority) {
        case WYSideslipGesturePriorityAutoSelection:
            self.currentGesturePriority = WYSideslipGesturePrioritySideslipFirst;
            break;
        case WYSideslipGesturePrioritySideslipFirst:
            self.currentGesturePriority = WYSideslipGesturePriorityNavigationBackFirst;
            break;
        case WYSideslipGesturePriorityNavigationBackFirst:
            self.currentGesturePriority = WYSideslipGesturePriorityAutoSelection;
            break;
    }
    
    // 重新创建导航栏按钮数组
    [self setupNavigationBar];
    
    // 使用封装的方法重置所有cell状态
    [self.tableView wy_resetAllVisibleCellsSideslipState];
    [self.tableView reloadData];
    
    wy_print(@"手势优先级: %@", [self gesturePriorityTitle]);
}

- (NSString *)gesturePriorityTitle {
    switch (self.currentGesturePriority) {
        case WYSideslipGesturePriorityAutoSelection:
            return @"手势:自动";
        case WYSideslipGesturePrioritySideslipFirst:
            return @"手势:侧滑优先";
        case WYSideslipGesturePriorityNavigationBackFirst:
            return @"手势:返回优先";
        default:
            return @"手势:自动";
    }
}

#pragma mark - UITableViewDelegate & UITableViewDataSource

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataSource.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
    
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    
    // 重置cell状态，防止重用问题
    [cell wy_resetSideslipState];
    
    // 配置侧滑功能
    NSString *direction = [self configureSideslipForCell:cell at:indexPath];
    
    // 显示功能状态
    NSString *longPullStatus = self.enableLongPull ? @"+长拉" : @"";
    NSString *gestureStatus = [self gestureStatusText];
    
    cell.textLabel.text = [NSString stringWithFormat:@"%@(%@%@%@)", self.dataSource[indexPath.row], direction, longPullStatus, gestureStatus];
    
    UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
    btn.backgroundColor = [UIColor.magentaColor colorWithAlphaComponent:0.25];
    btn.frame = CGRectMake((tableView.wy_width - 100)/2, 0, 100, 50);
    [btn addTarget:self action:@selector(didClickCellButton) forControlEvents:UIControlEventTouchUpInside];
    [cell.contentView addSubview:btn];
    
    return cell;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 60;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    wy_print(@"点击了第%ld个cell", (long)(indexPath.row + 1));
}

- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
    // 滑动tableView时关闭已侧滑的cell
    if ([scrollView isKindOfClass:[UITableView class]]) {
        UITableView *tableView = (UITableView *)scrollView;
        [tableView wy_closeCurrentOpenedSideslipCellIfNeeded];
    }
}

- (NSString *)configureSideslipForCell:(UITableViewCell *)cell at:(NSIndexPath *)indexPath {
    // 启用侧滑功能
    [cell wy_enableSideslip];
    
    // 设置手势优先级
    cell.wy_gesturePriority = self.currentGesturePriority;
    
    NSString *direction = @"";
    
    CGFloat leftSideslipWidth = 0;
    CGFloat rightSideslipWidth = 0;
    
    // 设置侧滑方向（可以根据需要调整）
    if (indexPath.row % 3 == 0) {
        cell.wy_sideslipDirection = WYTableViewSideslipDirectionLeft;
        direction = @"左侧侧滑";
        leftSideslipWidth = 80;
    } else if (indexPath.row % 3 == 1) {
        cell.wy_sideslipDirection = WYTableViewSideslipDirectionRight;
        direction = @"右侧侧滑";
        rightSideslipWidth = 120;
    } else {
        cell.wy_sideslipDirection = WYTableViewSideslipDirectionBoth;
        direction = @"两侧侧滑";
        leftSideslipWidth = 80;
        rightSideslipWidth = 120;
    }
    
    // 设置侧滑区域宽度
    cell.wy_leftSideslipWidth = leftSideslipWidth;
    cell.wy_rightSideslipWidth = rightSideslipWidth;
    
    // 配置长拉功能
    [self configureLongPullForCell:cell at:indexPath];
    
    // 设置自定义侧滑视图
    [self setupCustomSideslipViewForCell:cell at:indexPath];
    
    [cell wy_sideslipEvent:^(enum WYSideslipEventHandler event, enum WYTableViewSideslipDirection direction) {
        wy_print(@"event = %ld, direction = %ld", (long)event, (long)direction);
    }];
    
    return direction;
}

- (void)configureLongPullForCell:(UITableViewCell *)cell at:(NSIndexPath *)indexPath {
    if (self.enableLongPull) {
        cell.wy_enableLongPullAction = YES;
        cell.wy_longPullThreshold = 1.5;
        cell.wy_longPullHapticFeedback = YES;
        
        __weak typeof(self) weakSelf = self;
        __weak typeof(cell) weakCell = cell;
        
        [cell wy_sideslipLongPullHandlerWithProgress:^(CGFloat progress, WYTableViewSideslipDirection direction) {
            __strong typeof(weakSelf) strongSelf = weakSelf;
            __strong typeof(weakCell) strongCell = weakCell;
            if (!strongSelf || !strongCell) return;
            
            if (progress > 0) {
                // 通过tableView获取cell的当前indexPath
                NSIndexPath *currentIndexPath = [strongSelf.tableView indexPathForCell:strongCell];
                if (currentIndexPath) {
                    NSString *directionText = direction == WYTableViewSideslipDirectionLeft ? @"左侧" : @"右侧";
                    wy_print(@"第%ld行%@长拉进度: %.2f", (long)(currentIndexPath.row + 1), directionText, progress);
                }
            }
        } completion:^(WYTableViewSideslipDirection direction) {
            __strong typeof(weakSelf) strongSelf = weakSelf;
            __strong typeof(weakCell) strongCell = weakCell;
            if (!strongSelf || !strongCell) return;
            
            // 通过tableView获取cell的当前indexPath（最可靠的方式）
            NSIndexPath *currentIndexPath = [strongSelf.tableView indexPathForCell:strongCell];
            if (!currentIndexPath) {
                wy_print(@"❌ 无法获取cell的当前索引");
                return;
            }
            
            NSString *directionText = direction == WYTableViewSideslipDirectionLeft ? @"左侧" : @"右侧";
            wy_print(@"🎉 第%ld行%@长拉完成，执行对应事件！", (long)(currentIndexPath.row + 1), directionText);
            
            // 长拉完成后删除对应cell
            [strongSelf deleteCellAtIndexPath:currentIndexPath direction:direction];
        }];
    } else {
        cell.wy_enableLongPullAction = NO;
    }
}

- (void)setupCustomSideslipViewForCell:(UITableViewCell *)cell at:(NSIndexPath *)indexPath {
    // 左侧滑视图配置
    UIButton *leftButton = [self createSideslipButtonWithTitle:@"(左)删除" color:UIColor.systemRedColor indexPath:indexPath isLeft:YES];
    [cell wy_setSideslipView:leftButton direction:WYTableViewSideslipDirectionLeft];
    
    // 右侧滑视图配置
    UIButton *rightButton = [self createSideslipButtonWithTitle:@"(右)删除" color:UIColor.systemBlueColor indexPath:indexPath isLeft:NO];
    [cell wy_setSideslipView:rightButton direction:WYTableViewSideslipDirectionRight];
}

- (UIButton *)createSideslipButtonWithTitle:(NSString *)title color:(UIColor *)color indexPath:(NSIndexPath *)indexPath isLeft:(BOOL)isLeft {
    UIButton *button = [UIButton buttonWithType:UIButtonTypeSystem];
    [button setTitle:title forState:UIControlStateNormal];
    [button setTitleColor:UIColor.whiteColor forState:UIControlStateNormal];
    button.titleLabel.font = [UIFont systemFontOfSize:15];
    button.backgroundColor = color;
    
    // 使用更复杂的tag编码来区分左右按钮和行索引
    NSInteger buttonTag = indexPath.row * 100 + (isLeft ? 1 : 2);
    button.tag = buttonTag;
    
    [button addTarget:self action:@selector(handleButtonTap:) forControlEvents:UIControlEventTouchUpInside];
    
    return button;
}

- (NSString *)gestureStatusText {
    switch (self.currentGesturePriority) {
        case WYSideslipGesturePriorityAutoSelection:
            return @"";
        case WYSideslipGesturePrioritySideslipFirst:
            return @"+侧滑优先";
        case WYSideslipGesturePriorityNavigationBackFirst:
            return @"+返回优先";
        default:
            return @"";
    }
}

- (void)handleButtonTap:(UIButton *)sender {
    NSInteger buttonTag = sender.tag;
    NSInteger originalRowIndex = buttonTag / 100;
    BOOL isLeftButton = (buttonTag % 100) == 1;
    
    NSString *buttonType = isLeftButton ? @"左侧" : @"右侧";
    wy_print(@"点击了原始第 %ld 行的%@滑动区域按钮", (long)(originalRowIndex + 1), buttonType);
    
    // 通过按钮的superview找到对应的cell
    UITableViewCell *cell = [self findCellForButton:sender];
    if (cell) {
        // 通过tableView获取cell的当前indexPath（最可靠的方式）
        NSIndexPath *currentIndexPath = [self.tableView indexPathForCell:cell];
        if (currentIndexPath) {
            WYTableViewSideslipDirection direction = isLeftButton ? WYTableViewSideslipDirectionLeft : WYTableViewSideslipDirectionRight;
            [self deleteCellAtIndexPath:currentIndexPath direction:direction];
        } else {
            wy_print(@"❌ 无法找到按钮对应的cell当前索引");
        }
    }
}

// 通过按钮找到对应的cell
- (UITableViewCell *)findCellForButton:(UIButton *)button {
    UIView *view = button;
    while (view != nil) {
        if ([view isKindOfClass:[UITableViewCell class]]) {
            return (UITableViewCell *)view;
        }
        view = view.superview;
    }
    return nil;
}

- (void)deleteCellAtIndexPath:(NSIndexPath *)indexPath direction:(WYTableViewSideslipDirection)direction {
    if (indexPath.row >= self.dataSource.count) {
        wy_print(@"❌ 索引越界: %ld，数据源数量: %ld", (long)indexPath.row, (long)self.dataSource.count);
        return;
    }
    
    NSString *cellText = self.dataSource[indexPath.row];
    NSString *directionText = direction == WYTableViewSideslipDirectionLeft ? @"左侧" : @"右侧";
    
    wy_print(@"🗑️ 删除第%ld行 (%@): %@", (long)(indexPath.row + 1), directionText, cellText);
    
    // 先关闭侧滑
    UITableViewCell *cell = [self.tableView cellForRowAtIndexPath:indexPath];
    if (cell) {
        [cell wy_closeSideslipWithAnimated:NO];
    }
    
    // 执行删除动画
    [self.tableView performBatchUpdates:^{
        [self.dataSource removeObjectAtIndex:indexPath.row];
        [self.tableView deleteRowsAtIndexPaths:@[indexPath] withRowAnimation:UITableViewRowAnimationAutomatic];
    } completion:^(BOOL finished) {
        wy_print(@"✅ 删除完成，剩余%ld个cell", (long)self.dataSource.count);
    }];
}

- (void)didClickCellButton {
    wy_print(@"didClickCellButton");
}

- (void)dealloc {
    wy_print(@"WYTestSideslipCellController release");
}

/*
 #pragma mark - Navigation
 
 // In a storyboard-based application, you will often want to do a little preparation before navigation
 - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
 // Get the new view controller using [segue destinationViewController].
 // Pass the selected object to the new view controller.
 }
 */

@end

File: SwiftVerify/SwiftVerify/Examples/Controller/TableViewCell侧滑/WYTestSideslipCellController.swift
//
//  WYTestSideslipCellController.swift
//  SwiftVerify
//
//  Created by guanren on 2025/11/15.
//

import UIKit

class WYTestSideslipCellController: UIViewController {
    
    private let tableView = UITableView(frame: CGRect(x: 10, y: UIDevice.wy_navViewHeight, width: UIDevice.wy_screenWidth - 20, height: UIDevice.wy_screenHeight - UIDevice.wy_navViewHeight), style: .plain)
    private var dataSource: [String] = []
    
    // 控制开关
    private var enableLongPull = true
    private var currentGesturePriority: WYSideslipGesturePriority = .autoSelection
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Do any additional setup after loading the view.
        
        title = "侧滑功能验证"
        view.backgroundColor = .magenta
        
        setupNavigationBar()
        setupTableView()
        setupData()
        
        // 启用自动关闭侧滑功能（只需要调用一次）
        UITableView.wy_enableAutoCloseSideslip()
    }
    
    private func setupNavigationBar() {
        // 添加长拉功能开关
        let longPullButton = UIBarButtonItem(
            title: enableLongPull ? "长拉:开" : "长拉:关",
            style: .plain,
            target: self,
            action: #selector(toggleLongPull)
        )
        
        // 添加手势优先级切换
        let gesturePriorityButton = UIBarButtonItem(
            title: gesturePriorityTitle(),
            style: .plain,
            target: self,
            action: #selector(switchGesturePriority)
        )
        
        navigationItem.rightBarButtonItems = [longPullButton, gesturePriorityButton]
    }
    
    private func setupTableView() {
        tableView.delegate = self
        tableView.dataSource = self
        tableView.backgroundColor = .white
        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
        tableView.tableFooterView = UIView()
        
        view.addSubview(tableView)
    }
    
    private func setupData() {
        dataSource.removeAll()
        for i in 1...20 {
            dataSource.append("测试单元格 \(i)")
        }
        tableView.reloadData()
    }
    
    @objc private func toggleLongPull() {
        enableLongPull.toggle()
        navigationItem.rightBarButtonItems?[0].title = enableLongPull ? "长拉:开" : "长拉:关"
        
        // 使用封装的方法重置所有cell状态
        tableView.wy_resetAllVisibleCellsSideslipState()
        tableView.reloadData()
        
        WYLogManager.output("长拉功能: \(enableLongPull ? "开启" : "关闭")")
    }
    
    @objc private func switchGesturePriority() {
        switch currentGesturePriority {
        case .autoSelection:
            currentGesturePriority = .sideslipFirst
        case .sideslipFirst:
            currentGesturePriority = .navigationBackFirst
        case .navigationBackFirst:
            currentGesturePriority = .autoSelection
        }
        
        navigationItem.rightBarButtonItems?[1].title = gesturePriorityTitle()
        
        // 使用封装的方法重置所有cell状态
        tableView.wy_resetAllVisibleCellsSideslipState()
        tableView.reloadData()
        
        WYLogManager.output("手势优先级: \(gesturePriorityTitle())")
    }
    
    private func gesturePriorityTitle() -> String {
        switch currentGesturePriority {
        case .autoSelection:
            return "手势:自动"
        case .sideslipFirst:
            return "手势:侧滑优先"
        case .navigationBackFirst:
            return "手势:返回优先"
        }
    }
    
    deinit {
        WYLogManager.output("WYTestSideslipCellController release")
    }
    
    /*
     // MARK: - Navigation
     
     // In a storyboard-based application, you will often want to do a little preparation before navigation
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
     // Get the new view controller using segue.destination.
     // Pass the selected object to the new view controller.
     }
     */
    
}

extension WYTestSideslipCellController: UITableViewDelegate, UITableViewDataSource {
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataSource.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
        
        cell.selectionStyle = .none
        
        // 重置cell状态，防止重用问题
        cell.wy_resetSideslipState()
        
        // 配置侧滑功能
        let direction: String = configureSideslipForCell(cell, at: indexPath)
        
        // 显示功能状态
        let longPullStatus = enableLongPull ? "+长拉" : ""
        let gestureStatus = gestureStatusText()
        
        cell.textLabel?.text = dataSource[indexPath.row] + "(\(direction)\(longPullStatus)\(gestureStatus))"
        
        let btn = UIButton(type: .custom)
        btn.backgroundColor = UIColor.magenta.withAlphaComponent(0.25)
        btn.frame = CGRect(x: (tableView.wy_width - 100)/2, y: 0, width: 100, height: 50)
        btn.addTarget(self, action: #selector(didClickCellButton), for: .touchUpInside)
        cell.contentView .addSubview(btn)
        
        return cell
    }
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 60
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        WYLogManager.output("点击了第\(indexPath.row+1)个cell")
    }
    
    func scrollViewDidScroll(_ scrollView: UIScrollView) {
        // 滑动tableView时关闭已侧滑的cell
        if let tableView = scrollView as? UITableView {
            tableView.wy_closeCurrentOpenedSideslipCellIfNeeded()
        }
    }
    
    private func configureSideslipForCell(_ cell: UITableViewCell, at indexPath: IndexPath) -> String {
        // 启用侧滑功能
        cell.wy_enableSideslip()
        
        // 设置手势优先级
        cell.wy_gesturePriority = currentGesturePriority
        
        var direction: String = ""
        
        var leftSideslipWidth: CGFloat = 0
        var rightSideslipWidth: CGFloat = 0
        
        // 设置侧滑方向（可以根据需要调整）
        if indexPath.row % 3 == 0 {
            cell.wy_sideslipDirection = .left
            direction = "左侧侧滑"
            leftSideslipWidth = 80
        } else if indexPath.row % 3 == 1 {
            cell.wy_sideslipDirection = .right
            direction = "右侧侧滑"
            rightSideslipWidth = 120
        } else {
            cell.wy_sideslipDirection = .both
            direction = "两侧侧滑"
            leftSideslipWidth = 80
            rightSideslipWidth = 120
        }
        
        // 设置侧滑区域宽度
        cell.wy_leftSideslipWidth = leftSideslipWidth
        cell.wy_rightSideslipWidth = rightSideslipWidth
        
        // 配置长拉功能
        configureLongPullForCell(cell, at: indexPath)
        
        // 设置自定义侧滑视图
        setupCustomSideslipView(for: cell, at: indexPath)
        
        cell.wy_sideslipEventHandler { event, direction in
            WYLogManager.output("event = \(event), direction = \(direction)")
        }
        
        return direction
    }
    
    private func configureLongPullForCell(_ cell: UITableViewCell, at indexPath: IndexPath) {
        if enableLongPull {
            cell.wy_enableLongPullAction = true
            cell.wy_longPullThreshold = 1.5
            cell.wy_longPullHapticFeedback = true
            
            // 使用弱引用避免循环引用
            cell.wy_sideslipLongPullHandler(
                progress: { [weak self] progress, direction in
                    guard let self = self else { return }
                    if progress > 0 {
                        // 通过tableView获取cell的当前indexPath
                        if let currentIndexPath = self.tableView.indexPath(for: cell) {
                            let directionText = direction == .left ? "左侧" : "右侧"
                            WYLogManager.output("第\(currentIndexPath.row + 1)行\(directionText)长拉进度: \(String(format: "%.2f", progress))")
                        }
                    }
                },
                completion: { [weak self] direction in
                    guard let self = self else { return }
                    
                    // 通过tableView获取cell的当前indexPath（最可靠的方式）
                    guard let currentIndexPath = self.tableView.indexPath(for: cell) else {
                        WYLogManager.output("❌ 无法获取cell的当前索引")
                        return
                    }
                    
                    let directionText = direction == .left ? "左侧" : "右侧"
                    WYLogManager.output("🎉 第\(currentIndexPath.row + 1)行\(directionText)长拉完成，执行对应事件！")
                    
                    // 长拉完成后删除对应cell
                    self.deleteCell(at: currentIndexPath, direction: direction)
                }
            )
        } else {
            cell.wy_enableLongPullAction = false
        }
    }
    
    private func setupCustomSideslipView(for cell: UITableViewCell, at indexPath: IndexPath) {
        // 左侧滑视图配置
        let leftButton = createSideslipButton(title: "(左)删除", color: .systemRed, indexPath: indexPath, isLeft: true)
        cell.wy_setSideslipView(leftButton, for: .left)
        
        // 右侧滑视图配置
        let rightButton = createSideslipButton(title: "(右)删除", color: .systemBlue, indexPath: indexPath, isLeft: false)
        cell.wy_setSideslipView(rightButton, for: .right)
    }
    
    private func createSideslipButton(title: String, color: UIColor, indexPath: IndexPath, isLeft: Bool) -> UIButton {
        let button = UIButton(type: .system)
        button.setTitle(title, for: .normal)
        button.setTitleColor(.white, for: .normal)
        button.titleLabel?.font = UIFont.systemFont(ofSize: 15)
        button.backgroundColor = color
        
        // 使用更复杂的tag编码来区分左右按钮和行索引
        let buttonTag = indexPath.row * 100 + (isLeft ? 1 : 2)
        button.tag = buttonTag
        
        button.addTarget(self, action: #selector(handleButtonTap(_:)), for: .touchUpInside)
        
        return button
    }
    
    private func gestureStatusText() -> String {
        switch currentGesturePriority {
        case .autoSelection:
            return ""
        case .sideslipFirst:
            return "+侧滑优先"
        case .navigationBackFirst:
            return "+返回优先"
        }
    }
    
    @objc private func handleButtonTap(_ sender: UIButton) {
        let buttonTag = sender.tag
        let originalRowIndex = buttonTag / 100
        let isLeftButton = (buttonTag % 100) == 1
        
        let buttonType = isLeftButton ? "左侧" : "右侧"
        WYLogManager.output("点击了原始第 \(originalRowIndex + 1) 行的\(buttonType)滑动区域按钮")
        
        // 通过按钮的superview找到对应的cell
        if let cell = findCell(for: sender) {
            // 通过tableView获取cell的当前indexPath（最可靠的方式）
            if let currentIndexPath = tableView.indexPath(for: cell) {
                deleteCell(at: currentIndexPath, direction: isLeftButton ? .left : .right)
            } else {
                WYLogManager.output("❌ 无法找到按钮对应的cell当前索引")
            }
        }
    }
    
    // 通过按钮找到对应的cell
    private func findCell(for button: UIButton) -> UITableViewCell? {
        var view: UIView? = button
        while view != nil {
            if let cell = view as? UITableViewCell {
                return cell
            }
            view = view?.superview
        }
        return nil
    }
    
    private func deleteCell(at indexPath: IndexPath, direction: WYTableViewSideslipDirection) {
        guard indexPath.row < dataSource.count else {
            WYLogManager.output("❌ 索引越界: \(indexPath.row)，数据源数量: \(dataSource.count)")
            return
        }
        
        let cellText = dataSource[indexPath.row]
        let directionText = direction == .left ? "左侧" : "右侧"
        
        WYLogManager.output("🗑️ 删除第\(indexPath.row + 1)行 (\(directionText)): \(cellText)")
        
        // 先关闭侧滑
        if let cell = tableView.cellForRow(at: indexPath) {
            cell.wy_closeSideslip(animated: false)
        }
        
        // 执行删除动画
        tableView.performBatchUpdates({
            dataSource.remove(at: indexPath.row)
            tableView.deleteRows(at: [indexPath], with: .automatic)
        }, completion: { [weak self] _ in
            guard let self = self else { return }
            
            WYLogManager.output("✅ 删除完成，剩余\(self.dataSource.count)个cell")
        })
    }
    
    @objc func didClickCellButton() {
        WYLogManager.output("didClickCellButton")
    }
}
